src/edge_proxy/server.py

```diff
@@ -41,13 +41,29 @@ async def unknown_key_error(request, exc):
 @app.get("/health", response_class=ORJSONResponse, deprecated=True)
 @app.get("/proxy/health", response_class=ORJSONResponse)
 async def health_check():
-    with suppress(TypeError):
-        last_updated = datetime.now() - environment_service.last_updated_at
-        buffer = 30 * len(settings.environment_key_pairs)  # 30s per environment
-        if last_updated.total_seconds() <= settings.api_poll_frequency_seconds + buffer:
-            return ORJSONResponse(status_code=200, content={"status": "ok"})
+    last_updated_at = environment_service.last_updated_at
+    if not last_updated_at:
+        return HealthCheckResponse(
+            status_code=500,
+            status="error",
+            reason="environment document(s) not updated.",
+            last_successful_update=None,
+        )
 
-    return ORJSONResponse(status_code=500, content={"status": "error"})
+    if settings.health_check.count_stale_documents_as_failing:
+        buffer = (settings.health_check.grace_period_seconds or 30) * len(
```

Comment:
nit: we can add 30 as the default value in the settings, right?

Reply:
Yeah, I deliberated over this because the setting isn't used if `count_stale_documents_as_failing` is set to `False`, so it seemed a little odd for this setting to take a default value in that scenario. But I think it probably is cleaner that way.

Reply:
Done.